torch/lib/c10d/NCCLUtils.cpp

```diff
@@ -0,0 +1,31 @@
+#include <c10d/NCCLUtils.hpp>
+#include <mutex>
```

Comment:
Question: why do we need this header?

Reply:
This allows us to access `std::call_once` and `std::once_flag`